test/parallel/test-sqlite-database-sync.js

```diff
@@ -256,6 +274,15 @@ suite('DatabaseSync.prototype.exec()', () => {
     });
   });
 
+  test('throws if the URL does not have the file: scheme', (t) => {
```

Comment:
You should also check the happy path.

Can you try `URL("file://path/to/db.db?mode=ro` and verify that writes error out?

Passing a `file://` URL as a string or Buffer should also be supported.